README.md

```diff
@@ -1 +1,412 @@
+
+```objc
+// manual set up
+
+NSMutableArray *instructors = [[NSMutableArray alloc] init];
+[instructors addObjectsFromArray:@[@"Joe", @"Tim", @"Tom", @"Jim", @"Mark"]];
+
+NSLog(@"%@", instructors);
+```
+
+This will also print:
+
+```
+(
+    Joe,
+    Tim,
+    Tom,
+    Jim,
+    Mark
+)
+```
+It's likely that the `init` method actually calls the `initWithObjects:` method but with an empty list of objects to be added. At some point, a designated initializer **must** be called when creating a new instance; other initializers that are not defined as designated initializers **must** pass through a designated initializer. This is a way that the compiler helps to ensure that the instance will be functional at run time.
+
+**Note:** *While this is not strictly enforced by Objective-C itself, it is considered best practice—especially when writing Objective-C that will interface with anything written Swift which does hold strict rules about designated initializers.*
+
+### Declaring the Designated Initializer
+
+Since initializers *must* be public methods, they need to be declared in the `.h` header file. The rules for `init` family methods that are enforced by Apple via the compiler are that it:
+
+  * must be an instance method (`-`),
+  * must return an `instancetype`, and
+  * must begin with the word `init`.
+
+```objc
+- (instancetype)init...;
+```
+When writing any initializer that is not the default initializer `init`, it is common practice to continue the method name with the word `With` followed by an argument for every property into which the initializer is expected to pass a value. As convention, the arguments in the method declaration are typically named after the property to which it is intended to be assigned.
+
+So, assuming a custom class `FISWarship`:
+
+```obj
+// FISWarship.h
+
+#import <Foundation/Foundation.h>
+
+@interface FISWarship : NSObject
+
+@property (strong, nonatomic) NSString *shipName;
+@property (nonatomic) NSUInteger currentSpeedInKnots;
+@property (nonatomic) NSUInteger maximumSpeedInKnots;
+
+@end
+```
+
+The default initializer for a `FISWarship` class might be declared as:
+
+```objc
+- (instancetype)initWithShipName:(NSString *)shipName
+             currentSpeedInKnots:(NSUInteger)currentSpeedInKnots
+             maximumSpeedInKnots:(NSUInteger)maximumSpeedInKnots;
+```
+
+### Defining the Designated Initializer
+
+#### Autocomplete in Xcode
+
+Because designated initializers, in particular, can produce some ridiculously long method names in Objective-C, relying on autocomplete to reference the declared method elsewhere your in code is good practice to both save keystrokes and to defend against typos.
```

Comment:
I think I wrote that line before one of our discussions Tim, when I was still going off of the experience of writing the initializers for the data models in FISCardStreams, which were kinda crazy.